models/model.js

```diff
@@ -1,14 +1,38 @@
 const db = require('../db/connection.js')
 
-exports.selectCategories = () => {
-    return db.query('SELECT * FROM categories')
+exports.selectCategories = ({category}) => {
+    
+    let qryStr = 'SELECT * FROM categories'
+    let params = []
+    if (category !== undefined) {
+        qryStr += ' WHERE slug=$1'
+        params.push(category)
+    }
+    qryStr += ';'
+    return db.query(qryStr,params)
     .then(({rows}) => {
         return rows;
     })
 }
 
-exports.selectReviews = () => {
-    return db.query('SELECT owner,title,review_id,category,review_img_url,reviews.created_at,reviews.votes,designer,COUNT(comments.comment_id) AS comment_count FROM reviews LEFT JOIN comments USING (review_id) GROUP BY reviews.review_id ORDER BY created_at DESC;')
+exports.selectReviews = ({category, sort_by="created_at",order="DESC"}) => {
+    const acceptedSorts = ["owner","title","review_id","review_img_url","votes","designer","comment_count","created_at"]
+    const acceptedOrders = ["DESC","ASC"]
+    if (!acceptedSorts.includes(sort_by)) {
+        return Promise.reject({status:400,msg:"sort_by column not found in database"})
+    }
+    if (!acceptedOrders.includes(order.toUpperCase())) {
```

Comment:
To uppercase is a great solution to accept both upper and lowercase without extra code ✅